pkg/bucketcache/bucketcache.go

```diff
@@ -25,6 +27,24 @@ func (b *BucketCache[K, V]) AddToBucket(k K, val V) bool {
 	return b.addToCache(k, val, false)
 }
 
+func (b *BucketCache[K, V]) RemoveFromBucket(k K, val V) bool {
+	bucket, found := b.cache.Get(k)
+	if !found {
+		return false
+	}
+
+	newBucket := slices.DeleteFunc(bucket, func(v V) bool {
```

Comment:
From what I know, this function will modify the original slice, meaning there is no need to add it again to the cache.

Reply:
This also means, we need some kind of synchronization, as otherwise we could run into issues (aka concurrent modifications).

Reply:
It adds zeroes to the original slice and returns a new slice:

```go
func TestDeleteArrayElement(t *testing.T) {
	bucket := []int{1, 2, 3, 4, 5}

	newBucket := slices.DeleteFunc(bucket, func(v int) bool {
		return v == 3
	})

	require.Equal(t, []int{1, 2, 4, 5, 0}, bucket)
	require.Equal(t, []int{1, 2, 4, 5}, newBucket)
}
```

Reply:
Of course the length will be different 🤦 So all good haha 👍